### ISSUE TITLE
Disabling TLS causes redirect loop in Kibana

### ISSUE THREAD

**Issue by charith-elastic** [elastic/cloud-on-k8s#2065]
When running Kibana with the self-signed certificate disabled, users can't get past the login screen as it keeps redirecting back to itself. When the self-signed certificate is enabled, everything works as expected.

This is probably a case of requiring an extra configuration setting to make this work but nothing obvious leaps out from https://www.elastic.co/guide/en/kibana/current/settings.html.

**Comment by pebrc** [comment]
Did you by any chance reconfigure an existing Kibana to be non-HTTS? If so I wonder if the issue is to do with an existing session cookie with a different  Kibana encryption key https://www.elastic.co/guide/en/kibana/6.7/using-kibana-with-security.html#using-kibana-with-security in your browser. 

I you don't set the encryption key explicitly I think it will be regenerated on every configuration change that we roll out via ECK. I don't fully understand the mechanics here, but when I deleted my session cookie after running into the issue you described, I was able to log in again.

What i don't understand is why Kibana does not cleanly log you out in that situation.
